File: kotlin/ktfind/lib/src/main/kotlin/ktfind/FindConfig.kt
package ktfind

import java.nio.file.Paths

open class FindConfig {
    val defaultXFindConfigDir = Paths.get(System.getProperty("user.home"), ".config", "xfind").toString()
    val fileTypesPath: String
    val findOptionsPath: String
    val defaultFindSettingsPath: String

    init {
        fileTypesPath = "/filetypes.json"
        findOptionsPath = "/findoptions.json"
        val xFindConfigDir = System.getenv("XFIND_CONFIG_DIR") ?: defaultXFindConfigDir
        defaultFindSettingsPath = Paths.get(xFindConfigDir, "settings.json").toString()
    }
}

File: scala/scalafind/src/main/scala/scalafind/FindConfig.scala
package scalafind

import java.nio.file.Paths

class FindConfig {
  val defaultXFindConfigDir: String = Paths.get(System.getProperty("user.home"), ".config", "xfind").toString
  val fileTypesPath: String = "/filetypes.json"
  val findOptionsPath: String = "/findoptions.json"
  val xFindConfigDir: String = {
    if (System.getenv("XFIND_CONFIG_DIR") != null) {
      System.getenv("XFIND_CONFIG_DIR")
    } else {
      defaultXFindConfigDir
    }
  }
  val defaultFindSettingsPath: String = Paths.get(xFindConfigDir, "settings.json").toString
}
